syntax = "proto3";

package movies;

service MovieService {
  rpc GetMovieById (MovieRequest) returns (MovieResponse);
  rpc SearchMovies (SearchRequest) returns (MovieList);
}

message MovieRequest {
  int32 id = 1;
}

message SearchRequest {
  string query = 1;
}

message MovieResponse {
  int32 id = 1;
  string title = 2;
  int32 year = 3;
  string genre = 4;
  string description = 5;
  repeated string actors = 6;
}

message MovieList {
  repeated MovieResponse movies = 1;
}
